### PR TITLE
Mining headset upgrade - Callouts and volume boosters

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request
dreamseeker_SOPW0zf89I

Shift + middle clicking while wearing a mining headset will open a callout radial, after moving your mouse over one of the options a callout emote will appear where you pointed (No need to click on the radial button). Callouts have a 3 second cooldown to prevent spam and glow in the dark due to how dark lavaland is (normal point emotes do not)
There are 6 options: pointing, danger, attack, mine, defend and reposition. Your callouts are colored in your runechat color. This is done via a component so later if needed it could be added to other headsets/mobs/items. Callouts also can initiate basic mob orders, being a better way to command your minebots in combat.

Additionally, they also boost your speech back to normal levels in low-pressure environments, ensuring that your runechat is still nice and readable.

## Why It's Good For The Game

This would make coop mining much more enjoyable, as stopping to type mid-fight is more often than not a death sentence on lavaland. With arcmining's vents cooperating is actually beneficial, and I feel like we should incentivize miners to do it more often by providing them with tools for it.

## Changelog
:cl:
add: Mining headsets now allow you to make callouts via pointing. You can use them to communicate with fellow miners or order your army of bots and raptors!
add: Mining headsets keep your voice loud and clear in low-pressure environments (not vacuum!)
/:cl:

### CODE REVIEW

File: code/datums/components/callouts.dm [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+	RegisterSignal(atom_parent.loc, COMSIG_MOB_CLICKON, PROC_REF(on_click))
+	cur_user = atom_parent.loc
+
+/datum/component/callouts/RegisterWithParent()
+	. = ..()
+	RegisterSignal(parent, COMSIG_MOB_CLICKON, PROC_REF(on_click))
+	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, PROC_REF(on_equipped))
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, PROC_REF(on_dropped))
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, PROC_REF(on_examines))
```
Comment (MrMelbert) [inline code/datums/components/callouts.dm:50]: No `UnregisterFromParent`? Also parent call does nothing

File: code/datums/components/callouts.dm [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	if (!isnull(examine_text))
+		examine_list += examine_text
+
+/datum/component/callouts/proc/on_click(mob/user, atom/clicked_atom, list/modifiers)
+	SIGNAL_HANDLER
+
+	if (!LAZYACCESS(modifiers, SHIFT_CLICK) || !LAZYACCESS(modifiers, MIDDLE_CLICK))
+		return
+
+	if (world.time < last_callout_time + CALLOUT_COOLDOWN)
```
Comment (MrMelbert) [inline code/datums/components/callouts.dm:86]: Also use the cooldown defines. `COOLDOWN_DECLARE`, `COOLDOWN_START`, `COOLDOWN_FINISHED`

File: code/datums/components/callouts.dm [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		clicked_atom.balloon_alert(user, "callout is on cooldown")
```
Comment (MrMelbert) [inline code/datums/components/callouts.dm:87]:
```suggestion
		clicked_atom.balloon_alert(user, "callout is on cooldown!")
```

**Comment by Sadboysuss** [comment]
Also, I think the callout symbol should stay for longer so more people can see it and potentially be bigger, it's a bit hard to see if it's a sword or something else from afar

**Comment by SmArtKar** [comment]
good point

**Comment by mc-oofert** [comment]
spam pointing at something often gets the point across faster

**Comment by carlarctg** [comment]
make a special interaction with gold ore and stacks that makes you forcesay we're rich

**Comment by SomeoneYouProbablyKnow** [comment]
1 year before a sec callout system
3 years before a full removal of the talk button

**Comment by SmArtKar** [comment]
Per necro's suggestion you can now use callouts to command basic animals!

**Comment by JupiterJaeden** [comment]
Add to security headsets too plz….

Also it should definitely be toggleable.

**Comment by SmArtKar** [comment]
> Add to security headsets too plz….
> 
> Also it should definitely be toggleable.

image

security is a stationside job that interacts with crew, mining is a PVE job where standing in one place is certain death. There's a big difference - its intentionally miner only and I'm against adding it to other jobs/headsets

**Comment by SomeoneYouProbablyKnow** [comment]
> > Add to security headsets too plz….
> > Also it should definitely be toggleable.
> 
> image
> 

Nooooo i just found that picture as well. Ah well. i'll leave the links to similar sec calloout PRs for people to check out if they want it

https://github.com/tgstation/tgstation/pull/75853

https://github.com/tgstation/tgstation/pull/66578

**Comment by SmArtKar** [comment]
I've been told about #81940 making the intentional change to split crusher marks - but in practice that usually backfires, as it only creates confusion while removing potential of applying and bursting marks in quick succession from a skilled duo of miners

**Comment by SmArtKar** [comment]
> Add to security headsets too plz….
> 
> Also it should definitely be toggleable.

also yes, you can toggle them with ctrl click (because alt-click is reserved for loud mode)

**Comment by Admiral-Gigaton** [comment]
> Make it toggleable and by default on so you can still spam point on station

Or only work on lavaland but your idea is better for more opportunities to do bullshit with callouts

**Comment by ShizCalev** [comment]
conflicts

**Comment by optimumtact** [comment]
is typing in the game too hard for you?

:-1:

**Comment by CCC23** [comment]
WERE RICH!

**Comment by Loafin34** [comment]
sec callouts are for wusses that what don't believe in themselves of taking down the target and dying wordlessly 10 minutes in

**Comment by JupiterJaeden** [comment]
> > > Add to security headsets too plz….
> > > Also it should definitely be toggleable.
> > 
> > 
> > image
> 
> Nooooo i just found that picture as well. I wasn't fast enough. Ah well. i'll leave the links to similar sec calloout PRs for people to check out if they want it
> 
> #75853
> 
> #66578

Those are both VERY different because they say something on radio. This is just flavored pointing which could also be fun station side without being a serious balance concern.

**Review by Ghommie — APPROVED** [review]

**Pull request merged**